src/test/scheduler_tests.cpp

```diff
@@ -155,4 +155,45 @@ BOOST_AUTO_TEST_CASE(singlethreadedscheduler_ordered)
     BOOST_CHECK_EQUAL(counter2, 100);
 }
 
+BOOST_AUTO_TEST_CASE(mockforward)
+{
+    CScheduler scheduler;
+
+    int counter{0};
+    CScheduler::Function dummy = [&counter](){counter++;};
+
+    // schedule jobs for 2, 5 & 8 minutes into the future
+    int64_t min_in_milli = 60*1000;
+    scheduler.scheduleFromNow(dummy, 2*min_in_milli);
+    scheduler.scheduleFromNow(dummy, 5*min_in_milli);
+    scheduler.scheduleFromNow(dummy, 8*min_in_milli);
+
+    // check taskQueue
+    boost::chrono::system_clock::time_point first, last;
+    size_t num_tasks = scheduler.getQueueInfo(first, last);
+    BOOST_CHECK(num_tasks == 3);
+
+    boost::thread scheduler_thread(std::bind(&CScheduler::serviceQueue, &scheduler));
+
+    // bump the scheduler forward 5 minutes
+    scheduler.MockForward(boost::chrono::seconds(5*60));
+
+    // ensure scheduler has chance to process all tasks queued for before 1 ms from now.
+    scheduler.scheduleFromNow([&]{ scheduler.stop(false); }, 1);
```

Comment:
nit, just capture `&scheduler`.

Reply:
I'm curious- is this for style or is there a difference in behavior? 
from what I can tell the behavior should be the same?
docs: "& (implicitly capture the used automatic variables by reference)" 
compiler explorer: https://godbolt.org/z/QfeAVF

Reply:
I think & is perfectly clear, it's just for style.

I agree in principle if a lambda is multiple lines and has arguments it's usually better to explicitly capture, but I think implicit capture is totally reasonable here.

Reply:
Yes, should be same, however in L163 you already explicitly capture `&counter` so I though I would like to do the same here.

Reply:
updated